jOOQ/src/main/java/org/jooq/impl/InsertImpl.java

```diff
@@ -604,6 +609,27 @@ public final InsertImpl doNothing() {
         return onDuplicateKeyIgnore();
     }
 
+    @Override
+    public final InsertImpl onConflictOnConstraint(Constraint constraint) {
+        getDelegate().onConflictOnConstraint(constraint);
+        return this;
+    }
+
+    @Override
+    public final InsertImpl onConflictOnConstraint(Name constraint) {
+        onConflictOnConstraint(constraint(constraint));
+        return this;
+    }
+
+    @Override
+    public final InsertImpl onConflictOnConstraint(UniqueKey<R> constraint) {
+        if (constraint.getName() == null)
+            throw new IllegalStateException("UniqueKey's name is not specified");
```

Comment:
Good idea. I'll switch this to `StringUtils.isEmpty()` (as empty names are possible too, and are also illegal here). Also, `IllegalArgumentException` seems more appropriate in this particular case

Reply:
... I've noticed just now that the exception is also thrown from `InsertQueryImpl` (which makes sense, of course). So, to remove repetition of code, I will change the above `InsertImpl` logic to simply delegate to `InsertQueryImpl` and let the latter handle the plumbing / data type conversion.

Reply:
I agree, using `StringUtils.isEmpty()` in this case is more appropriate then checking on `null`